homeassistant/components/tuya/climate.py

```diff
@@ -35,14 +36,20 @@
 from .const import DOMAIN, TUYA_DISCOVERY_NEW, DPCode
 
 TUYA_HVAC_TO_HA = {
-    "auto": HVAC_MODE_HEAT_COOL,
+    "auto": HVAC_MODE_AUTO,
     "cold": HVAC_MODE_COOL,
     "freeze": HVAC_MODE_COOL,
     "heat": HVAC_MODE_HEAT,
     "hot": HVAC_MODE_HEAT,
     "manual": HVAC_MODE_HEAT_COOL,
     "wet": HVAC_MODE_DRY,
     "wind": HVAC_MODE_FAN_ONLY,
+    
+    #Tuya HY368 radiator
+    "eco": HVAC_MODE_HEAT,
```

Comment:
eco doesn't have to apply to heat only?

The same applies to the ones below.

Reply:
HY368 can only turn the heater on/off. It cannot make it start to cool.

Reply:
Sure, but this logic is not exclusive for your HY368 device.

Reply:
@frenck **If** there are other devices that have these modes, an error currently occurs when they are selected. This script is buggy and needs to be fixed.

I don't own devices other than the HY368, I don't know if they can do anything other than heating in those modes. Do you own such a device?

Reply:
I think a compromise would be a change such that if the mode is out of the current list, the mode is `HVAC_MODE_HEAT_COOL`?
is certainly not acceptable the way it currently works - it doesn't work at all.

Reply:
@frenck du You have better sollution for #62839 than mine?

Reply:
I'm not sure what you are asking for this at this point?

Reply:
this pull request is the only existing solution to the problem. Either reject it and the bug will remain, or accept it and the bug will be gone. Unless you know of another solution that fixes this bug?

Reply:
I am confused? The heat mode is incorrect in some cases. Are you suggesting merging a know bug to fix your known bug?

Reply:
This integration does not support some modes. How do you want to fix this?

My suggestion is to use `HVAC_MODE_HEAT_COOL` in this place

before my fix - it does not work
after my fix - it works